src/14_legendre.jl

```diff
@@ -0,0 +1,372 @@
+        y->y)                               # P_1(y)
+end
+function legendrep(n::Integer, m::Integer, x)
+
+    # special case: m=0  =>  P_n^m = P_n
+    # Legendre polynomial P_n(x) can be evaluated outside of [-1,1].
+    # Thus, we use that routine before checking x interval for associated Legendre polynomial P_n^m(x).
+    (0 == m) && return legendrep(n, x)
+
+    # general argument checks
+    (n <  0)          && throw(DomainError(n, "must be non-negative"))
+    (m <  0 || m > n) && throw(DomainError(m, "must be in the range m = 0,..,n"))
+
+    # step 1: compute P_m^m(x)
+    p = one(x)                                      # = P_0^0(x)
+    for k = 1:m
+        p_prev  = p
+        p       = -(2k-1) * sqrt(1-x^2) * p_prev    # = P_k^k(x)
+    end                                             # on output: p = P_m^m(x)
+    (n == m) && return p
+
+    # step 2: compute P_{m+1}^m(x)
+    p_prev  = p                                     # = P_m^m(x)
+    p       = (2m+1) * x * p_prev                   # = P_{m+1}^m(x)
+
+    # step 3: compute P_n^m(x)
+    for k = m+2:n
+        p_prev, p_prev_prev = p, p_prev
+        p = (2k-1)/(k-m)*x*p_prev - (k+m-1)/(k-m)*p_prev_prev       # = P_k^m(x)
+    end                                             # on output: p = P_n^m(x)
+
+    p
+end
+
+@doc raw"""
+    legendreq(n[, m], x)
+
+Evaluate the Legendre function of second kind ``Q_n(x)`` of integer degree ``n`` at position
+``x``.
+If the order ``m`` is supplied then the associated Legendre function of second kind
+``Q_n^{(m)}(x)`` is computed.
+Their definitions are given by
+
+```math
+\begin{align}
+    Q_n^{(m)}(x)
+    &=
+    (-1)^m (1-x^2)^\frac{m}{2} \frac{\mathrm{d}^m}{\mathrm{d}x^m} \left( Q_n(x) \right)
+    \,, \quad m \in \mathbb{N}_0 \,, \; x \in (-1,1)
+    \\
+    Q_n(x)
+    &=
+    \begin{cases}
+        \frac{1}{2} \log \frac{1+x}{1-x}
+        & \text{for} \quad n = 0
+        \\
+        P_1(x) Q_0(x) - 1
+        & \text{for} \quad n = 1
+        \\
+        \frac{2n-1}{n} x Q_{n-1}(x) - \frac{n-1}{n} Q_{n-2}(x)
+        & \text{for} \quad n \geq 2
+        \,.
+    \end{cases}
+\end{align}
+```
+where ``P_1`` is the Legendre polynomial of first kind and degree one.
+The functions diverge at the interval boundaries
+``\lim_{x\rightarrow \pm 1} |Q_n(x)| = \infty``.
+
+External links: [DLMF - Legendre function](https://dlmf.nist.gov/14.7.E2),
+[DLMF - associated Legendre function](https://dlmf.nist.gov/14.7.E9),
+[Wikipedia - Legendre function](https://en.wikipedia.org/wiki/Legendre_functions#Legendre_functions_of_the_second_kind_(Qn)),
+[Wikipedia - associated Legendre function](https://en.wikipedia.org/wiki/Legendre_function#Associated_Legendre_functions_of_the_second_kind).
+"""
+function legendreq(n::Integer, x)
+    (0 > n)      && throw(DomainError(n, "must be non-negative"))
+    (1 < abs(x)) && throw(DomainError(x, "must be in the range [-1,1]"))
+
+    if x == 1
+        return Inf
```

Comment:
not type-stable.   e.g. if `x` is single-precision this will return a double-precision `Inf`

Reply:
i thought about it too. in my eyes the problem was rather how to treat the case if x is of integer type, i.e. ``x=1``.? what should the return type/value be? 
i thought to just return ``Inf::Float64`` in all cases. It's not like one can do calculations with it anyways..

open for any ideas...

Reply:
If x is an integer it should return a float, ie ` float(typeof(x)) `

Reply:
pushed updated version in commit [02557a8](https://github.com/JuliaMath/SpecialFunctions.jl/pull/175/commits/02557a8549fb33cf4f517a09d767994bed063359).
if `x` is of floating type it will return `Inf` of same type, e.g. for `x::Float32` it returns `Inf::Float32`. if `x` is of integer type it will return `Inf::Float`.